src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -486,7 +493,7 @@ public int getMaxRequestsPerHost() {
 
     @DataBoundSetter
     public void setConnectTimeout(int connectTimeout) {
-        this.connectTimeout = connectTimeout;
+        this.connectTimeout = Math.max(DEFAULT_CONNECT_TIMEOUT_SECONDS, connectTimeout);
```

Comment:
For such cases it would be polite to add form validation warning admins that attempts to set this to a lower value will be ignored.